### ISSUE TITLE
Version 0.5.0 does not work as update from 0.4.1 (Periodic errors: "use of closed network connection")

### ISSUE THREAD

**Issue by jacotec** [strukturag/nextcloud-spreed-signaling#271]
I've updated te HPB from 0.4.1 to 0.5.0 and unfortunately none of my NC instances are working anymore.

In the NC talk settings, the HPB is successfully recognized as a running HPB and the correct Github version hash is displayed.

But no chat can connect either for chatting nor for calls. The message input field and the "Start call" button stays gray.

When opening a room and every 5 seconds the syslog is filling up with these error messages:

```
Jun  8 14:21:33 talk signaling[4524]: client.go:282: Client from 10.200.0.1 has RTT of 21 ms (21.827794ms)
Jun  8 14:21:33 talk signaling[4524]: hub.go:822: Invalid message {Marshaler:<nil> Unmarshaler:<nil> Id: Type: Hello:<nil> Bye:<nil> Room:<nil> Message:<nil> Control:<nil> Internal:<nil> TransientData:<nil>} from 10.200.0.1: type missing
Jun  8 14:21:35 talk signaling[4524]: client.go:303: Error reading from 10.200.0.1: read tcp 127.0.0.1:8080->127.0.0.1:40040: use of closed network connection
Jun  8 14:21:36 talk signaling[4524]: client.go:303: Error reading from 10.200.0.1: read tcp 127.0.0.1:8080->127.0.0.1:40042: use of closed network connection
Jun  8 14:21:36 talk signaling[4524]: client.go:282: Client from 10.200.0.1 has RTT of 32 ms (32.174717ms)
Jun  8 14:21:36 talk signaling[4524]: hub.go:822: Invalid message {Marshaler:<nil> Unmarshaler:<nil> Id: Type: Hello:<nil> Bye:<nil> Room:<nil> Message:<nil> Control:<nil> Internal:<nil> TransientData:<nil>} from 10.200.0.1: type missing
Jun  8 14:21:37 talk signaling[4524]: client.go:282: Client from 10.200.0.1 has RTT of 21 ms (21.7479ms)
Jun  8 14:21:37 talk signaling[4524]: hub.go:822: Invalid message {Marshaler:<nil> Unmarshaler:<nil> Id: Type: Hello:<nil> Bye:<nil> Room:<nil> Message:<nil> Control:<nil> Internal:<nil> TransientData:<nil>} from 10.200.0.1: type missing
Jun  8 14:21:39 talk signaling[4524]: client.go:303: Error reading from 10.200.0.1: read tcp 127.0.0.1:8080->127.0.0.1:40044: use of closed network connection
Jun  8 14:21:40 talk signaling[4524]: client.go:303: Error reading from 10.200.0.1: read tcp 127.0.0.1:8080->127.0.0.1:40046: use of closed network connection
Jun  8 14:21:40 talk signaling[4524]: client.go:282: Client from 10.200.0.1 has RTT of 21 ms (21.67138ms)
Jun  8 14:21:40 talk signaling[4524]: hub.go:822: Invalid message {Marshaler:<nil> Unmarshaler:<nil> Id: Type: Hello:<nil> Bye:<nil> Room:<nil> Message:<nil> Control:<nil> Internal:<nil> TransientData:<nil>} from 10.200.0.1: type missing
Jun  8 14:21:41 talk signaling[4524]: client.go:282: Client from 10.200.0.1 has RTT of 22 ms (22.707271ms)
Jun  8 14:21:41 talk signaling[4524]: hub.go:822: Invalid message {Marshaler:<nil> Unmarshaler:<nil> Id: Type: Hello:<nil> Bye:<nil> Room:<nil> Message:<nil> Control:<nil> Internal:<nil> TransientData:<nil>} from 10.200.0.1: type missing

```

Seems I need to roll back to 0.4.1 ... what's wrong with 0.5.0?

**Comment by fancycode** [comment]
How did you build 0.5.0? The messages `hub.go:822: Invalid message ...` indicate a problem with the easyjson marshalling/unmarshalling code (see #180).

**Comment by jacotec** [comment]
@fancycode As usual (my config lives in /etc/signaling, so I'm safe to build from scratch each time):

- Stop service "signaling"
- Remove the dir /etc/nextcloud-spreed-signaling
- cd /etc
- git clone --branch v0.5.0 https://github.com/strukturag/nextcloud-spreed-signaling.git
- cd nextcloud-spreed-signaling
- make build

Then I start the service "signaling" again.

Go is: go version go1.17.11 linux/amd64

Doing exactly the same with "git clone --branch 0.4.1 ..." and all is working again.

**Comment by fancycode** [comment]
Could you please provide the output of `make build`? Also, does `make test` succeed?

**Comment by jacotec** [comment]
@fancycode Trying the fix from #180 makes it work!

Strange: I have no remains in the source directory as I'm fully removing it before the checkout. Nevertheless, running a "make clean" before "make build" solves it.

Maybe the readme.md should be updated accordingly, which also says only "make build"?

**Comment by fancycode** [comment]
> Maybe the readme.md should be updated accordingly, which also says only "make build"?

Well, this should not be necessary after a fresh checkout (and works without on CI), so I'm wondering why you had to run it manually. The files removed by `make clean` are generated by `make build` and cleaning should only be required after a partial/failed build.
